data_masks/redactor.go

```diff
@@ -102,16 +107,22 @@ func NewRedactor(
 		}
 	}()
 
+	running := &atomic.Bool{}
+	running.Store(true)
+
 	return &Redactor{
 		SensitiveDataKeys:          sets.NewSet(config.SensitiveKeys...),
 		SensitiveDataValuePatterns: sensitiveDataRegex,
 		userConfig:                 activeUserConfig,
 		exitChannel:                exitChannel,
+		running:                    running,
 	}, nil
 }
 
 func (o *Redactor) StopPeriodicUpdates() {
-	close(o.exitChannel)
+	if o.running.Swap(false) {
+		close(o.exitChannel)
+	}
```

Comment:
sync.Once would be better for accomplishing this.

Reply:
In #84.